src/main/java/org/owasp/esapi/waf/ESAPIWebApplicationFirewallFilter.java

```diff
@@ -60,448 +58,436 @@
-                try {
-                    inputStream.close();
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            }
-        }
-    }
-
-    public AppGuardianConfiguration getConfiguration() {
-        return appGuardConfig;
-    }
-
-    /**
-     * 
-     * This function is invoked at application startup and when the
-     * configuration file polling period has elapsed and a change in the
-     * configuration file has been detected.
-     * 
-     * It's main purpose is to read the configuration file and establish the
-     * configuration object model for use at runtime during the
-     * <code>doFilter()</code> method.
-     */
-    public void init(FilterConfig fc) throws ServletException {
-
-        /*
-         * This variable is saved so that we can retrieve it later to re-invoke
-         * this function.
-         */
-        this.fc = fc;
-
-        logger.debug(Logger.EVENT_SUCCESS, ">> Initializing WAF");
-        /*
-         * Pull logging file.
-         */
-
-        // Demoted scope to a local since this is the only place it is
-        // referenced
-        String logSettingsFilename = fc.getInitParameter(LOGGING_FILE_PARAM);
-
-        String realLogSettingsFilename = fc.getServletContext().getRealPath(logSettingsFilename);
-
-        if (realLogSettingsFilename == null || (!new File(realLogSettingsFilename).exists())) {
-            throw new ServletException(
-                    "[ESAPI WAF] Could not find log file at resolved path: " + realLogSettingsFilename);
-        }
-
-        /*
-         * Pull main configuration file.
-         */
-
-        configurationFilename = fc.getInitParameter(CONFIGURATION_FILE_PARAM);
-
-        configurationFilename = fc.getServletContext().getRealPath(configurationFilename);
-
-        if (configurationFilename == null || !new File(configurationFilename).exists()) {
-            throw new ServletException(
-                    "[ESAPI WAF] Could not find configuration file at resolved path: " + configurationFilename);
-        }
-
-        /*
-         * Find out polling time from a parameter. If none is provided, use the
-         * default (10 seconds).
-         */
-
-        String sPollingTime = fc.getInitParameter(POLLING_TIME_PARAM);
-
-        if (sPollingTime != null) {
-            pollingTime = Long.parseLong(sPollingTime);
-        } else {
-            pollingTime = DEFAULT_POLLING_TIME;
-        }
-
-        /*
-         * Open up configuration file and populate the AppGuardian configuration
-         * object.
-         */
-
-        FileInputStream inputStream = null;
-
-        try {
-            String webRootDir = fc.getServletContext().getRealPath("/");
-            inputStream = new FileInputStream(configurationFilename);
-            appGuardConfig = ConfigurationParser.readConfigurationFile(inputStream, webRootDir);
-            DOMConfigurator.configure(realLogSettingsFilename);
```

Comment:
What the heck was the point of this call, anyway? Just to see if the log settings file was a legitimate log4j.xml type configuration file? Or was it expected that since originally ESAPI was using Log4J as it's default logger, tat this just allowed the ESAPI WAF to log to a different file than the rest of ESAPI?

I know that the WAF unit tests all pass without this call to DOMConfigurator, but OTOH I don't have warm fuzzies about the thoroughness of its tests.

Reply:
I think it's assuming that WAF has some specific settings that are required, and that those settings cannot be added directly to a pre-existing configuration.
I'm not sure of the use case exactly.

I'm not really sure what to do with this but to dissect the WAF test suites and try to make sense of it.  I can look at this, but I have a feeling it will take a bit to get my head around what's going on.

Reply:
Hey @nahsra  - Can you give us any insight into the ESAPI WAF use case here -- of what this `Log4J DOMConfigurator` was used for? I know if was a long time ago, but you were the original author, so I'm hoping you remember because we are trying to remove the dependency on Log4J 1. here. But OTOH, I don't want to break anything for any clients who might be making advanced use of the WAF and I'm not fully confident that the WAF unit tests are complete enough to test this particular unknown (to us) functionality.

Reply:
If I had to guess, it had something to do with the WAF originally being as "drop-in" utility that wanted to operate on its own logging settings, rather than rely on the app it was included in. 

I don't see any harm in nuking it and letting the WAF portion use ESAPIs typical logging strategy.

Reply:
Okay, thanks for taking the time to look at it Arshan.